### PR TITLE
returning false if buffer full for exchange manager

### PR DESCRIPTION
**Pull request by vincentberenz**
changes are related to the exchange manager:

- exchange manager used previously a segment id and an object id. This was just to only using a segment id, the object id being created automatically. 

- functions for adding new commands returns false if the commands were not added because the buffered queue was full

- added some dirty hack that content of empty interprocess queues are purged. This is strange because it should not be possible to remove items from an empty queue. But if this is not done, the programs randomly hang.

- added exchange manager unit tests

### CODE REVIEW

File: include/shared_memory/exchange_manager_consumer.hpp [hunk 1]
```diff
@@ -94,6 +94,10 @@ namespace shared_memory {
      *  from the exchange queue. For debugging purposes
      */
     int nb_char_read();
+
+    bool producer_queue_empty() const;
```
Comment (MaximilienNaveau) [inline include/shared_memory/exchange_manager_consumer.hpp]: Semantic idea, I do not want you to change anything.
I would personally call them `bool is_XXX()` for me it is a little clearer that the method return a bool and perform an internal check.
Again this comment is not really important.

File: include/shared_memory/internal/exchange_manager_memory.hxx [hunk 2]
```diff
@@ -127,13 +127,52 @@ Exchange_manager_memory<Serializable,QUEUE_SIZE>::Exchange_manager_memory( std::
...
   object_id_producer_ = object_id+"_producer";
   object_id_consumer_ = object_id+"_consumer";
   object_id_status_ = object_id+"_status";
   produced_ = segment_.find_or_construct<producer_queue>(object_id_producer_.c_str())();
   consumed_ = segment_.find_or_construct<consumer_queue>(object_id_consumer_.c_str())();
   segment_id_ = segment_id;
   values_ = new char[serializable_size_];
+
+  // dev note: no idea why this could be required
+  weird_purge();
```
Comment (MaximilienNaveau) [inline include/shared_memory/internal/exchange_manager_memory.hxx]: The comment is miss-leading. Please write doxygen comment about it in the declaration. And reference the definition and the usage to the declaration.

File: include/shared_memory/exchange_manager_consumer.hpp (hunk 1 above)
Comment (vincentberenz) [reply include/shared_memory/exchange_manager_consumer.hpp]: +1, will do

**Comment by vincentberenz** [comment]
addressed the comments and merged

**Pull request merged**